vim/merlin/autoload/merlin.py

```diff
@@ -644,15 +644,16 @@ def vim_case_analysis():
 
     vim_type_reset()
 
-def vim_type_enclosing():
+def vim_type_enclosing_at_pos(bufnr, to_line, to_col):
     global enclosing_types
     global current_enclosing
+    if bufnr != vim.current.buffer.number:
+        return '{}'
```

Comment:
This can disappear (and the function can be renamed to lose the `vim_` prefix) if the other change I proposed is implemented.